api/views.py

```diff
@@ -17,7 +17,10 @@ def post(self, request):
         if serializer.is_valid():
             order = serializer.validated_data
 
-            Client(order).create_delivery()
+            delivery = Client(order).create_delivery()
+
+            if 'error' not in delivery.to_dict():
```

Comment:
maybe we should return something if there is an error? if so... make the if be the error catching... and else: the not error logic

Reply:
I could check if what's return is an Error object instead of a Delivery but that wouldn't change much in terms of coupling. We would still be checking the type of thing we're dealing with. Besides, this still needs to be worked on, as I said in the description.